backend/src/apiserver/storage/experiment_store.go

```diff
@@ -217,6 +236,116 @@ func (s *ExperimentStore) DeleteExperiment(id string) error {
 	return nil
 }
 
+func (s *ExperimentStore) ArchiveExperiment(expId string) error {
+	// ArchiveExperiment results in
+	// 1. The experiment getting archived
+	// 2. All the runs in the experiment getting archived no matter what previous storage state they are in
+	sql, args, err := sq.
+		Update("experiments").
+		SetMap(sq.Eq{
+			"StorageState": api.Experiment_STORAGESTATE_ARCHIVED.String(),
+		}).
+		Where(sq.Eq{"UUID": expId}).
+		ToSql()
+	if err != nil {
+		return util.NewInternalServerError(err,
+			"Failed to create query to archive experiment %s. error: '%v'", expId, err.Error())
+	}
+
+	updateRunsSql, updateRunsArgs, err := sq.
+		Update("run_details").
+		SetMap(sq.Eq{
+			"StorageState": api.Run_STORAGESTATE_ARCHIVED.String(),
+		}).
+		Where(sq.Eq{"ExperimentUUID": expId}).ToSql()
```

Comment:
Be aware that `ExperimentUUID` was added very recently. Old runs may not have this field.

Reply:
Thanks for the heads up!
I'll then switch to use resource references  for a more consistent and reliable filtering (since we'll need to do similar filter in Job table as well and Job table doesn't have experiment uuid column)